src/App.vue

```diff
@@ -249,11 +255,52 @@ export default class App extends Mixins(mixins.TransactionMixin, NodeErrorMixin)
     }
   }
 
+  private applyTheme(isDark: boolean) {
+    setTheme(isDark ? Theme.DARK : Theme.LIGHT);
+    updatePipTheme();
+    tmaSdkService.updateTheme();
+  }
+
+  private detectSystemTheme() {
+    this.prefersDarkScheme = window.matchMedia('(prefers-color-scheme: dark)');
+
+    // This is needed when change not in TG Mini App
+    this.handleThemeChange = async (e: MediaQueryListEvent) => {
```

Comment:
I can't find the place where this method is resets to null, so maybe it's better to define it on data initialization step?